src/setfit/utils.py

```diff
@@ -49,6 +49,18 @@
 }
 
 
+def default_hp_space_optuna(trial) -> Dict[str, float]:
+    from transformers.integrations import is_optuna_available
+
+    assert is_optuna_available(), "This function needs Optuna installed: `pip install optuna`"
+    return {
+        "learning_rate": trial.suggest_float("learning_rate", 1e-6, 1e-4, log=True),
+        "num_epochs": trial.suggest_int("num_epochs", 1, 5),
```

Comment:
What do you think about adding `num_iterations` (i.e. the number of contrastive learning triples to generate) to the default space, since this hyperparameter is quite important for SetFit?

E.g. we could include:

```
num_iterations: trial.suggest_categorical("num_iterations", [5,10,20])